src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -234,6 +234,11 @@ protected function finalizeValue($value)
         }
 
         foreach ($this->children as $name => $child) {
+            if ($child->isDeprecated()) {
+                $msg = sprintf('The child node "%s" at path "%s" is deprecated.', $name, $this->getPath());
```

Comment:
I think that customizing the message would solve the thing (with mandatory place holders like `%node_name%` and `%node_path%`. A bit like what is done on deprecating services, which has a mandatory `%service_id%` placeholder (or something like that).